executor/src/core.rs

```diff
@@ -64,21 +66,27 @@ where
                 rx_reconfigure,
                 rx_subscriber,
             }
-            .run()
+            .run(restored_messages)
             .await
             .expect("Failed to run core")
         })
     }
 
     /// Main loop listening to new certificates and execute them.
-    async fn run(&mut self) -> SubscriberResult<()> {
+    async fn run(&mut self, restored_messages: Vec<ConsensusOutput>) -> SubscriberResult<()> {
         let _next_certificate_index = self
             .execution_state
             .load_next_certificate_index()
             .await
             .expect("Failed to load execution indices from store");
 
-        // TODO: Replay certificates from the store.
+        // Process any messages that were restored after a restart, by order of sequence.
+        for message in restored_messages
+            .into_iter()
+            .sorted_by(|a, b| a.consensus_index.cmp(&b.consensus_index))
+        {
+            self.execute_certificate(&message).await?;
+        }
```

Comment:
Is there a scenario where this loop can take a very long time?

Reply:
The longer the executor was down, the greater the amount of restored messages there could be. However, as I understand, it is essential that ensure they are executed in order, right?